openapi/workflow_execution_service.openapi.yaml

```diff
@@ -542,6 +546,22 @@ components:
           type: string
         state:
           $ref: '#/components/schemas/State'
+    RunListStatus:
+      title: RunListStatus
+      required:
+        - run_id
+      type: object
+      properties:
+        run_id:
+          type: string
+        state:
+          $ref: '#/components/schemas/State'
+        start_time:
+          type: string
+          description: When the run started executing, in ISO 8601 format "%Y-%m-%dT%H:%M:%SZ"
+        end_time:
+          type: string
+          description: When the run stopped executing (completed, failed, or cancelled), in ISO 8601 format "%Y-%m-%dT%H:%M:%SZ"
```

Comment:
1. This is essentially a copy of `RunStatus` that has two more _optional_ properties. What it does, essentially, is to encourage implementers to return a run's start and end time and to use the suggested properties `start_time` and `end_time` for that purpose, which is great. But I don't see why it needs another model for that, because:

   - Given that `additionalProperties` is not set to `false` in the `RunStatus` schema, implementers _could_ choose to return these info for `GET /runs/{id}/status` as well, and if they did, it would be nice if they used the same properties.
   - Using just one model would still allow implementers to return a different response for `GET /runs` and `GET /runs/{id}/status` if they so wish. There really isn't any benefit of having two different models here.

   If we ever end up _requiring_ different properties for the `GET /runs` and `GET /runs/{id}/status` responses, we can still add another model. Or, if possible, extend the `RunStatus` model with additional required parameters through the use of `allOf`.

2. Also, given that all that adding optional properties to a non-closed, extensible model does is to describe how certain information is to be preferentially returned if it is returned at all, I don't see why we wouldn't also include the other properties we have discussed, such as `workflow_type` and `tags`. Implementers could still choose _not_ to return them. But if they do, at least chances are high that they are going to make use of the suggested property names rather than coming up with their own, which opens up a path to require some or all of these properties in a future breaking version of the spec with the smallest possible impact.

Reply:
@uniqueg I fully agree with you for the first point. I think for now we can use a the `RunStatus` object for stuffing all of this information in.

As per the additional properties, the removal of `tags` and other fields actually came up in a discussion with the other leads. We felt like we wanted to make some incremental change on this ticket and it seemed like the extra properties (tags, workflow_type etc) were slightly more controversial and could be addressed at a later time. IMO adding the time is a huge win already if we get this pushed.

We COULD reuse the `required` property and basically add `run_id`, `state`, `end_time` and `start_time` as required  and then also add `tags`, `workflow_type`, `workflow_type_version`, `workflow_engine`, `workflow_engine_version` and `workflow_url` as optional fields. This would increase the footprint of the minimal representation

Reply:
Alright, sounds good :)